src/Microsoft.Dnx.ApplicationHost/Program.cs

```diff
@@ -229,19 +229,29 @@ private Task<int> ExecuteMain(DefaultHost host, string applicationName, string[]
             {
                 assembly = host.GetEntryPoint(applicationName);
             }
-            // Compilation exceptions either throw FileNotFound or FileLoad exceptions and may change
-            // from version to version. It's safest to catch both types of exceptions
-            catch (Exception ex) when (ex is FileLoadException || ex is FileNotFoundException)
+            catch (Exception ex)
             {
-                if (ex.InnerException is ICompilationException)
+                // Try to find compilation exception recursively to supress its stack
+                var innerException = ex;
+                do
                 {
-                    throw SuppressStackTrace(ex.InnerException);
+                    if (innerException is ICompilationException)
+                    {
+                        throw SuppressStackTrace(innerException);
+                    }
+                    innerException = innerException.InnerException;
```

Comment:
Could the `innerException == null` here (therefore cause `NullRef`)? There is nothing guard this from happening. Or is there any implicit rules there that the captured the exception must have an inner exception?